otherlibs/systhreads/st_stubs.c

```diff
@@ -294,8 +292,8 @@ static void save_runtime_state(void)
   th->gc_regs_buckets = Caml_state->gc_regs_buckets;
   th->exn_handler = Caml_state->exn_handler;
   th->async_exn_handler = Caml_state->async_exn_handler;
+  caml_get_local_arenas_and_save_local_sp(th->current_stack);
```

Comment:
I think this is being executed purely for the side-effect of setting:
```
th->current_stack->local_sp = Caml_state->local_sp;
```
and if so, it would probably clearer to write that directly.